src/settingsActions.js

```diff
@@ -17,30 +17,49 @@ const settingsActions = Action.createActionsFromNames([
     'showSnackbarMessage',
 ]);
 
+const saveLocalizedAppearanceSetting = (d2, key, value, locale) => {
+    const api = d2.Api.getApi();
+    const localeSuffix = locale ? `&locale=${locale}` : '';
+    const url = `/systemSettings/${key}?value=${value}${localeSuffix}`;
+
+    return api.post(url)
+        .then(() => {
+            settingsActions.showSnackbarMessage(d2.i18n.getTranslation('settings_updated'));
+        })
+        .catch((err) => {
+            log.warn('Failed to save localized setting:', err);
```

Comment:
Might be nice to show the (or an) error to the user. Or will they be notified otherwise that the update has failed?

Reply:
I see that the original implementation did this as well, so could consider that out of scope for this PR.